packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -275,6 +278,18 @@ class CupertinoSearchTextField extends StatefulWidget {
   /// {@macro flutter.widgets.Focus.focusNode}
   final FocusNode? focusNode;
 
+  /// {@macro flutter.material.textfield.onTap}
+  final VoidCallback? onTap;
+
+  /// {@macro flutter.widgets.editableText.autocorrect}
+  final bool autocorrect;
+
+  /// Disables the text field when false.
+  ///
+  /// Text fields in disabled states have a light grey background and don't
+  /// respond to touch events.
+  final bool? enabled;
```

Comment:
Well this may not be as easy as passing `enabled` to `CupertinoTextField` here. When a search field is disabled the prefix/suffix widgets may need changing too (for instance, if there's a "clear" button then it may look different when `enabled` is false, or it shouldn't appear at all).

Reply:
It should be clear now that setting `enabled` to false, the built-int search button and `suffixIcon` will not respond to touch events.

Reply:
Should the clear button grey out if you set `enabled` to false?

Reply:
Currently it defaults to grey (regardless of `enabled` being true/false). I'm not familiar with the iOS style but I would prefer to leave this as-is so people users can customize this behavior.